akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -213,6 +236,9 @@ object HttpEntity {
     def withContentType(contentType: ContentType): Default =
       if (contentType == this.contentType) this else copy(contentType = contentType)
 
+    def withSizeLimit(maxBytes: Long): Default =
+      copy(data = data withAttributes Attributes(SizeLimit(maxBytes, Some(contentLength))))
```

Comment:
Only checking the contentLength late means we may introduce strange effects if someone uses 

```scala
entity.withSizeLimit(abc).copy(contentLength = newCL, data = entity.data.transform(xyz))
```

What's the reason not to fail early if possible?